### ISSUE TITLE
Why is SIZE called in FtpExists.NoCheck mode?

### ISSUE THREAD

**Issue by armind13** [robinrodricks/FluentFTP#421]
**FTP OS:** Windows

**FTP Server:** FileZilla Server 0.9.60 beta

**Computer OS:** Windows

I want to save files with overwriting but I doesn't want to use `FtpExists.Overwrite`, because this usage will call extra SIZE and DELL commands. When I use `FtpExists.NoCheck` mode, library call SIZE before STOR. Why? Why library does not just call STOR for save and overwriting? As I know [standart supports](https://tools.ietf.org/html/rfc959) saving with overwritng for only STOR command.
I upload files use this code:
```
using (var ftpClient = new FluentFTP.FtpClient("localhost", 21, "user1", "password1"))
{
    ftpClient.OnLogEvent = (level, s) => { Console.WriteLine($"{level}: {s}"); };
    ftpClient.Connect();

    var paths = Directory.GetFiles(@"C:\dev\ftp_src");
    foreach (var path in paths)
    {
        var content = File.ReadAllBytes(path);
        var remotePath = Path.Combine("in", Path.GetFileName(path));
        ftpClient.Upload(content, remotePath, FtpExists.NoCheck);
    }
}
```

I don't want to use SIZE command for minimize command traffic.

**Logs :**

```
C:/dev/FluentFtpTest/FtpClient/bin/Debug/FtpClient.exe
Verbose: >         Connect()
Info: Status:   Connecting to ::1:21
Info: Response: 220 Please visit https://filezilla-project.org/
Verbose: Response: 220-FileZilla Server 0.9.60 beta
Response: 220-written by Tim Kosse (tim.kosse@filezilla-project.org)
Info: Status:   Detected FTP server: FileZilla
Info: Command:  USER user1
Info: Response: 331 Password required for user1
Info: Command:  PASS ***
Info: Response: 230 Logged on
Info: Command:  FEAT
Info: Response: 211 End
Verbose: Response: 211-Features:
Response: MDTM
Response: REST STREAM
Response: SIZE
Response: MLST type*;size*;modify*;
Response: MLSD
Response: UTF8
Response: CLNT
Response: MFMT
Response: EPSV
Response: EPRT
Info: Status:   Text encoding: System.Text.UTF8Encoding
Info: Command:  OPTS UTF8 ON
Info: Response: 202 UTF8 mode is always enabled. No need to send this command.
Info: Command:  SYST
Info: Response: 215 UNIX emulated by FileZilla
Info: Status:   Auto-detected UNIX listing parser
Verbose: >         Upload("in\first.txt", NoCheck, False)
Verbose: >         OpenWrite("in\first.txt", Binary)
Info: Command:  TYPE I
Info: Response: 200 Type set to I
Verbose: >         GetFileSize("in\first.txt")
Info: Command:  SIZE in/first.txt
Info: Response: 213 12
Verbose: >         OpenPassiveDataStream(AutoPassive, "STOR in/first.txt", 0)
Info: Command:  EPSV
Info: Response: 229 Entering Extended Passive Mode (|||64178|)
Info: Status:   Connecting to ::1:64178
Info: Command:  STOR in/first.txt
Info: Response: 150 Opening data channel for file upload to server of "/in/first.txt"
Verbose: Status:   Disposing FtpSocketStream...
Info: Response: 226 Successfully transferred "/in/first.txt"
Verbose: >         Upload("in\second.txt", NoCheck, False)
Verbose: >         OpenWrite("in\second.txt", Binary)
Verbose: >         GetFileSize("in\second.txt")
Info: Command:  SIZE in/second.txt
Info: Response: 213 14
Verbose: >         OpenPassiveDataStream(AutoPassive, "STOR in/second.txt", 0)
Info: Command:  EPSV
Info: Response: 229 Entering Extended Passive Mode (|||56221|)
Info: Status:   Connecting to ::1:56221
Info: Command:  STOR in/second.txt
Info: Response: 150 Opening data channel for file upload to server of "/in/second.txt"
Verbose: Status:   Disposing FtpSocketStream...
Info: Response: 226 Successfully transferred "/in/second.txt"
Verbose: >         Dispose()
Verbose: Status:   Disposing FtpClient object...
Info: Command:  QUIT
Info: Response: 221 Goodbye
Verbose: Status:   Disposing FtpSocketStream...
Verbose: Status:   Disposing FtpSocketStream...
```

**Comment by robinrodricks** [comment]
Can you check if this works...

https://we.tl/t-IGvJ7HrEcb

**Comment by robinrodricks** [comment]
Released in 26.0.0.

https://www.nuget.org/packages/FluentFTP/26.0.0
